cylc/flow/config.py

```diff
@@ -844,12 +845,19 @@ def _check_circular(self):
         """Check for circular dependence in graph."""
         start_point_string = (
             self.cfg['visualization']['initial cycle point'])
+        raw_graph = self.get_graph_raw(start_point_string,
+                                       stop_point_string=None)
+        if (len(raw_graph) > self.CHECK_CIRCULAR_LIMIT and not (
+                getattr(self.options, 'check_circular', False) or
+                self._is_validate(is_strict=True))):
```

Comment:
The `--strict` option is going as soon as we decide on [something](https://github.com/cylc/cylc-flow/issues/3866),  can make it so this is just activated by `--check-circular`.

Reply:
Ah ok